File: find_collections.py
import os, re

src = r'd:\Billing-app\billing-app\src'
collections = set()

p1 = re.compile(r'collection\(db,\s*["\'](\w+)["\']')
p2 = re.compile(r'doc\(db,\s*["\'](\w+)["\']')
p3 = re.compile(r'adminDb\.collection\(["\'](\w+)["\']')

for root, dirs, files in os.walk(src):
    for f in files:
        if f.endswith(('.ts', '.tsx')):
            try:
                with open(os.path.join(root, f), 'r', encoding='utf-8', errors='ignore') as fh:
                    content = fh.read()
                    for m in p1.finditer(content): collections.add(m.group(1))
                    for m in p2.finditer(content): collections.add(m.group(1))
                    for m in p3.finditer(content): collections.add(m.group(1))
            except: pass

for c in sorted(collections):
    print(c)

File: find_collections.js
const fs = require('fs');
const path = require('path');

const src = 'd:/Billing-app/billing-app/src';
const cols = new Set();

function walk(dir) {
  const items = fs.readdirSync(dir, { withFileTypes: true });
  for (const i of items) {
    const p = path.join(dir, i.name);
    if (i.isDirectory()) {
      walk(p);
    } else if (i.name.endsWith('.ts') || i.name.endsWith('.tsx')) {
      const c = fs.readFileSync(p, 'utf8');
      const patterns = [
        /collection\(db,\s*["'](\w+)["']/g,
        /doc\(db,\s*["'](\w+)["']/g,
        /adminDb\.collection\(["'](\w+)["']/g,
        /collection\(["'](\w+)["']\)/g,
      ];
      for (const pat of patterns) {
        for (const m of c.matchAll(pat)) {
          cols.add(m[1]);
        }
      }
    }
  }
}

walk(src);
[...cols].sort().forEach(c => console.log(c));
